test/src/org/apache/jmeter/protocol/http/util/TestHTTPArgument.java

```diff
@@ -94,4 +99,138 @@ public void testEncoding() throws Exception {
             assertEquals("", arg.getEncodedName());
             assertEquals("\00\01\07", arg.getEncodedValue());
         }
+        
+        @Test
+        public void testShift_JISEncoding() throws Exception {
+            HTTPArgument arg;
+            arg = new HTTPArgument("name.?", "value_ here", false, "Shift_JIS");
+            assertEquals("name.?", arg.getName());
+            assertEquals("value_ here", arg.getValue());
+            assertEquals("name.%3F", arg.getEncodedName());
+            assertEquals("value_+here", arg.getEncodedValue());
+            // Show that can bypass encoding:
+            arg.setAlwaysEncoded(false);
+            assertEquals("name.?", arg.getEncodedName());
+            assertEquals("value_ here", arg.getEncodedValue());
+            
+            // The sample does not use a valid encoding
+            arg = new HTTPArgument("name.?", "value_ here", true, "Shift_JIS");
```

Comment:
Use a new method for the next case.
I would advise you to have a look at the spock test cases. I think the tabular data format in those tests will be a great fit for these tests.

Reply:
Could you please suggest me how I can run individual spock test as I can't find run as junit test for spock test in eclipse. It looks like I am missing some configuration settings.

Reply:
Review comments incorporated. I have made one private method having parameter encoding to test all encoding passed by public test method. Please check